### PR TITLE
Add SDKMan Java Version Recipe

### PR DESCRIPTION
**Pull request by MBoegers**
Implements an OpenRewrite recipe that automatically manages `.sdkmanrc` files for Java projects using SDKMan.

## What it does:

- Scans multi-project builds to detect Java versions from compilation unit markers
- Creates or updates `.sdkmanrc` files in the project root
- Uses the highest Java version found across all source sets (with Java 8 fallback)
- Handles existing `.sdkmanrc` files by updating only the Java version line

## Key features:

- Smart detection: Analyzes JavaVersion markers from compilation units
- Highest version selection: Automatically picks the most recent Java version in use
- Non-destructive updates: Preserves existing `.sdkmanrc` content (maven, gradle, etc.)
- Comprehensive tests: Covers multiple scenarios including edge cases

## Example:

* Java 17 in main source set
* Java 21 in test source set
* Result: java=21 in `.sdkmanrc`

This recipe helps standardize Java version management across projects using SDKMan's `.sdkmanrc` configuration files.

### CODE REVIEW

File: src/main/java/com/yourorg/UseSdkManJavaVersion.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public Accumulator getInitialValue(ExecutionContext ctx) {
+        return new Accumulator();
+    }
+
+
+    public static class Accumulator {
+        boolean sdkmanrcExists = false;
```
[the remark this answers is not in the document]
Comment (timtebeek) [reply src/main/java/com/yourorg/UseSdkManJavaVersion.java:36]: No need for atomic there indeed; not aware of any plans for parallel recipes. The reason you sometimes see AtomicBoolean is that folks skip the wrapping Accumulator class.

**Review by timtebeek — APPROVED** [review]
